hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeConfiguration.java

```diff
@@ -338,16 +340,24 @@ public void setWaitOnAllFollowers(boolean val) {
   )
   private int rocksdbMaxFileNum = ROCKSDB_LOG_MAX_FILE_NUM_DEFAULT;
 
-  @Config(key = "rocksdb.delete_obsolete_files_period",
-      defaultValue = "6h", timeUnit = MICROSECONDS,
+  @Config(key = "rocksdb.delete-obsolete-files-period",
```

Comment:
Are we changing the config name? replacing `_` with `-`, Even above ``ROCKSDB_DELETE_OBSOLETE_FILES_PERIOD_MICRO_SECONDS_KEY`` has ``_``, I am not sure but changing config names must be an incompatible change right?

Reply:
The main purpose of Replacing _ with - is to be consistent with other rocksdb property name style. The property is a recently added, should not introduce any compatibility issue.

Reply:
@ChenSammi do we not need to change _ with - here in the value of this variable as well
``` public static final String
      ROCKSDB_DELETE_OBSOLETE_FILES_PERIOD_MICRO_SECONDS_KEY =
      "hdds.datanode.rocksdb.delete_obsolete_files_period";
```

Reply:
> The property is recently added, should not introduce any compatibility issue.

@ChenSammi, please cherry-pick this to the branch `ozone-1.3` to avoid change of config property name between releases.